### PR TITLE
Features/rhpm oa02

### PR DESCRIPTION
**Pull request by bharathkeerthi**
approval history for opening and candidate

### CODE REVIEW

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs [hunk 1]
```diff
@@ -58,6 +62,33 @@ public List<Approvals> GetAllApprovals()
...
+                    join users in _context.Users on ApprovalTransactionDetail.CreatedBy equals users.UserId
+
+                    where (ApprovalTran.EntityId == entityId)
+
+                    select new ApprovalTransactionDetailsDTO
+                    {
+                        ApprovalActionName = ApprovalAction.ApprovalActionName,
+                        FirstName = users.FirstName,
+                        LastName = users.LastName,
+                        CreatedDate = ApprovalTransactionDetail.ModifiedDate.Value,
```
Comment (salilsaini868) [inline api/RS.API/RS.Data/Logic/ApprovalRepository.cs]: Rename CreatedDate to ActionPerformedDate

File: api/RS.API/RS.Service/Logic/OpeningManagerService.cs [hunk 2]
```diff
@@ -143,9 +143,10 @@ public IResult GetOpeningsCorrespondingToLoggedUser(Guid userId)
                     MapPrimaryandSecondarySkills(openingViewModel, opening);
                     openingViewModel.CreatedDate = opening.CreatedDate.ToString("MM/dd/yyyy");
                     openingViewModel.ModifiedDate = opening.ModifiedDate.HasValue ? opening.ModifiedDate.Value.ToString("MM/dd/yyyy") : string.Empty;
```
Comment (salilsaini868) [inline api/RS.API/RS.Service/Logic/OpeningManagerService.cs]: don't format date in backend, format in frontent using pipe.

File: web/src/app/admin/skills/skills.component.html [hunk 3]
```diff
@@ -15,7 +15,7 @@
                                     </div>
                                     <div class="col-md-9 col-sm-9 col-xs-12">
                                         <input autofocus type="text" class="full-input" id="skills" id="name" name="skillName" [(ngModel)]="skillsModel.name" #skillName="ngModel" required minlength="3"
-                                            maxlength="20">
+                                            maxlength="20" (blur)="onBlurMethod()">
```
Comment (salilsaini868) [inline web/src/app/admin/skills/skills.component.html]: use proper method names every where.

File: web/src/app/admin/skills/skills.component.ts [hunk 4]
```diff
@@ -50,9 +52,22 @@ export class SkillsComponent implements OnInit {
...
   }
+  checkDuplicate(enteredSkill){
+    for(let skill of this.skills){
+      if(skill == enteredSkill){
+        this.duplicate = true
+      }
+    }
+    this.duplicate = false
+  }
+  onBlurMethod(){
```
Comment (salilsaini868) [inline web/src/app/admin/skills/skills.component.ts]: use proper method names.

File: web/src/app/admin/skills/skills.component.ts [hunk 5]
```diff
@@ -50,9 +52,22 @@ export class SkillsComponent implements OnInit {
...
+    this.skillsModel;
+    this.displayMessage.showWarning("skills name already exist ");
```
Comment (salilsaini868) [inline web/src/app/admin/skills/skills.component.ts]: here use label for displaying message, remove toaster from here

File: web/src/app/approval/approvalhistory.component.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Input() entityId: any;
+
+    constructor(private router: Router, private msgService: DisplayMessageService,private approvalServiceApp: ApprovalServiceApp, private route: ActivatedRoute) {
+    }
+    ngOnInit() {
+        this.getApprovalDetails();
+    }
+    getApprovalDetails() {
+        debugger;
```
Comment (salilsaini868) [inline web/src/app/approval/approvalhistory.component.ts]: remove debugger every where

**Review by salilsaini868 — CHANGES_REQUESTED** [review]
Mentioned changes should need to be implement every where.

File: api/RS.API/RS.Service/Logic/OpeningManagerService.cs (hunk 2 above)
Comment (bharathkeerthi) [reply api/RS.API/RS.Service/Logic/OpeningManagerService.cs]: done

File: web/src/app/approval/approvalhistory.component.ts (hunk 6 above)
Comment (bharathkeerthi) [reply web/src/app/approval/approvalhistory.component.ts]: done

File: api/RS.API/RS.Data/Logic/ApprovalRepository.cs (hunk 1 above)
Comment (bharathkeerthi) [reply api/RS.API/RS.Data/Logic/ApprovalRepository.cs]: done

File: web/src/app/admin/skills/skills.component.html (hunk 3 above)
Comment (bharathkeerthi) [reply web/src/app/admin/skills/skills.component.html]: done

File: web/src/app/admin/skills/skills.component.ts (hunk 4 above)
Comment (bharathkeerthi) [reply web/src/app/admin/skills/skills.component.ts]: done

File: web/src/app/admin/skills/skills.component.ts (hunk 5 above)
Comment (bharathkeerthi) [reply web/src/app/admin/skills/skills.component.ts]: done

**Review by salilsaini868 — APPROVED** [review]

**Pull request merged**